### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by mfherbst**

### CODE REVIEW

**Comment by mfherbst** [comment]
@maxscheurer Are you sure we can just include Psi4 files (LGPL) into a Boost Software Licence project?

**Comment by mfherbst** [comment]
@maxscheurer For now I think this can go in.

**Comment by mfherbst** [comment]
Just uploaded: https://get.adc-connect.org/libtensorlight/libtensorlight-3.0.0.tar.gz Needs linux and openblas naturally.

**Comment by maxscheurer** [comment]
Mhm maybe we can also use normal options and remove it? 🤷🏼‍♂️

**Review by maxscheurer — APPROVED** [review]

**Comment by maxscheurer** [comment]
I want to fix some CMake stuff wrt cmake configs before merge.

**Comment by maxscheurer** [comment]
Also tests don't compile on macOS:
```
/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/strings.h:73:7: note: candidate found by name lookup is 'index'
char    *index(const char *, int) __POSIX_C_DEPRECATED(200112L);
         ^
../libtensor/core/index.h:111:7: note: candidate found by name lookup is 'libtensor::index'
class index : public sequence<N, size_t> {
      ^
../tests/dense_tensor/tod_dirsum_test.C:299:5: error: reference to 'index' is ambiguous
    index<2> ia1, ia2;
```

**Comment by mfherbst** [comment]
@maxscheurer I changed a lot, can you try again?

by the way libxm now works and MPI in theory as well.

**Comment by maxscheurer** [comment]
Compiling the tests still fails because of the error above...

**Comment by maxscheurer** [comment]
I'm working on a fix.

**Comment by maxscheurer** [comment]
Nice, now a have a yummy template error:
```
In file included from ../libtensor/core/impl/allocator.C:1:
In file included from ../libtensor/core/impl/allocator_impl.h:7:
../libtensor/core/impl/../allocator_init.h:16:12: error: no template named 'allocator_wrapper'; did you mean 'allocator_wrapper_i'?
    static allocator_wrapper<T, AllocatorImpl> a(aimpl);
           ^~~~~~~~~~~~~~~~~
           allocator_wrapper_i
../libtensor/core/impl/../allocator.h:18:7: note: 'allocator_wrapper_i' declared here
class allocator_wrapper_i {
      ^
In file included from ../libtensor/core/impl/allocator.C:1:
In file included from ../libtensor/core/impl/allocator_impl.h:7:
../libtensor/core/impl/../allocator_init.h:16:12: error: too many template arguments for class template 'allocator_wrapper_i'
    static allocator_wrapper<T, AllocatorImpl> a(aimpl);
           ^                    ~~~~~~~~~~~~~~
../libtensor/core/impl/../allocator.h:18:7: note: template is declared here
class allocator_wrapper_i {
```

**Comment by mfherbst** [comment]
@maxscheurer Could you do me the honours and fix my macos workflow here ;)

**Comment by maxscheurer** [comment]
I'm on it... 😄

**Comment by maxscheurer** [comment]
Would be best not to commit anything right now such that our stuff doesn't clash...

**Comment by maxscheurer** [comment]
The CMake config stuff is also broken, so I'm going to fix everything now 😄

File: libtensor/core/impl/xm_allocator.h [hunk 1]
```diff
@@ -1,234 +1,223 @@
 #ifndef LIBTENSOR_XM_ALLOCATOR_H
```
Comment (mfherbst) [inline libtensor/core/impl/xm_allocator.h]: Sorry to be picky here, but can you turn clang-format off when committing this file.

Comment (maxscheurer) [reply libtensor/core/impl/xm_allocator.h]: Ahh... sorry 😦

**Comment by maxscheurer** [comment]
Yay 🚀 🎉

**Pull request merged**